File: Codes/Kas_ilke.c
#include <stdio.h>

int main()
{
     printf("-------------------1---------------------\n");
    printf("For loops\n");
    int arr[] ={0,1,2,3,4};
    int i;
    int j;
    printf("0 value is considered as false in the second expression\n");
    for(i = 4; arr[i]; i--)
    {
        printf("arr[%d] : %d\n",i,arr[i]);
    }
    printf("\n");
    i = 0;
    printf("First and the third expressions are absent:\n");
    int arr3[5] = {0,1,2,3,4};
    for( ; arr3[i] <2 ; )
    {
          printf("arr3[%d] : %d\n",i,arr3[i]);
          i++;
    }
    printf("Multiple statements for first and third expressions: \n");
    int arr4[5] = {0,1,2,3,4};
    for(i= 3, j = 2; i < 5 && j <5; arr4[i] +=1, arr4[j] *=2, i++,j++)
    {
        printf("arr4[%d] : %d, arr4[%d]: %d",i,j,arr4[i],arr4[j]);
    }
    printf("For loop without body: \n");
    int arr5[5] = {0,1,2,3,4};
    for(i = 1; arr5[i] < 3; arr5[i] *= 2, printf("arr5[%d]: %d\n",i,arr5[i]));
    
    printf("While and do wihle loops: \n");
    printf("0 value is considered as false: \n");
    int arr6[5] = {0,1,2,3,4};
    i = 4;
    while(arr6[i])
    {
        printf("arr6[%d]:%d\n",i,arr6[i]);   
        i--;
    }
    printf("While statement without body: \n");
    while(arr6[i]);
    printf("\n");
    printf("While loop vs. do while loop\n");
    while(arr6[i])
    {
        printf("printed in while\n");
    }
    do{
        printf("printed in do-while\n");
    }while(arr6[i]);
    printf("Iteration of arrays:\n");
    int array[] = {10,34,2,8,65,20};
    int size = 6;
    printf("Iteration with for:\n ");
    for(i = 0; i < size ; i++)
    {
        printf("array[%d] : %d\n", i, array[i]);
        
    }
    printf("-------------------2---------------------\n");
    printf("Iteration with while:\n ");
    i = 0;
    while(i < size)
    {
         printf("array[%d] : %d\n", i, array[i]);
         i++;
    }
     printf("Iteration with do while:\n ");
       i = 0;
    do
    {
         printf("array[%d] : %d\n", i, array[i]);
         i++;
    }while(i < size);
     printf("-------------------3---------------------\n");
     int array2 [] = {1,3,5,7};
     size = 4;
     int currentIndex = 0;
     int currentNum = array2[currentIndex];
     printf("current num: %d\n",currentNum);
     int nextNum = array2[++currentIndex];
     printf("next num: %d\n",nextNum);
     nextNum = array2[++currentIndex];
      printf("next num: %d\n",nextNum);
     nextNum = array2[++currentIndex];
      printf("next num: %d\n\n",nextNum);
      
      for(currentIndex = 0; currentIndex < size; currentIndex ++)
      {
            printf("next num: %d\n",array2[currentIndex]);
      }
    return 0;
}

File: Codes/Kas_ilke.php
<!DOCTYPE html>
<html>
<body>

<?php
function main()
{
    echo "---------1----------</br>";
    echo "For Loops </br>";
    $arr = array(0,1,2,3,4);
    for($x = 0 ; $x < sizeof($arr); $x++)
    {
        echo "arr[",$x,"]: ",$arr[$x],"</br>";
    }
    $x = 0;
    echo "First and last expressions are absent</br>";
     for( ; $x < sizeof($arr); )
    {
        echo "arr[",$x,"]: ",$arr[$x],"</br>";
        $x++;
    }
    $i = 0;
    echo "For loop without body</br>";
     for( ; $i < sizeof($arr); $i++ );
     echo "i: ",$i,"</br>";
    echo "While and do while loops</br>";
    $j = 0;
    while($j < sizeof($arr))
    {
        echo "arr[",$j,"]: ",$arr[$j],"</br>";
        $j++;
    }
    echo "While loop without body</br>";
    while($j < 0);
    echo "While loop vs. do while loop</br>";
    while($arr[0])
    {
        echo "Print in while</br>";
    }
    do
    {
        echo "Print in do while</br>";
    }while($arr[0]);
    echo "For Each Loops </br>";
    $list = array("milk","bread","egg","tomato");
    foreach($list as $item)
    {
        echo "$item </br>";
    }
    echo "Associative Arrays</br>";
    $person_list = array("İlke" => "milk", "Zeynep" => "bread", "Aysu" => "egg");
    foreach($person_list as $person => $item)
    {
        echo "$person => $item </br>";
    }
    echo "Strings</br>";
    $str = "Ilke";
    for($i = 0; $i <strlen($str); $i++)
    {
        echo "$str[$i] </br>";
    }
    $numbers = array(1,3,5,7,9);
    echo "cur:",current($numbers). "</br>";
    echo "next: ",next($numbers)."</br>";
    echo "cur: ",current($numbers). "</br>";
    echo "next: ",next($numbers)."</br>";
    echo "cur: ",current($numbers). "</br>";
    echo "prev: ",prev($numbers)."</br>";
    echo "cur: ",current($numbers). "</br>";
    echo "cur:",reset($numbers). "</br>";
    while($cur_val = next($numbers))
        echo "next num: ", $cur_val."</br>";
        
    $person_list2 = array("İlke" => "milk", "Zeynep" => "bread", "Aysu" => "egg");
    echo "cur:",current($person_list2). "</br>";
    echo "next: ",next($person_list2)."</br>";
    echo "cur: ",current($person_list2). "</br>";
    echo "next: ",next($person_list2)."</br>";
    echo "cur: ",current($person_list2). "</br>";
    echo "prev: ",prev($person_list2)."</br>";
    echo "cur: ",current($person_list2). "</br>";
    echo "cur:",reset($person_list2). "</br>";
    while($cur_val = next($person_list2))
        echo "next num: ", $cur_val."</br>";
  
    
}
main();
?>

</body>
</html>
